backends/fs/android/android-saf-fs.cpp

```diff
@@ -542,6 +544,59 @@ bool AndroidSAFFilesystemNode::createDirectory() {
 	return true;
 }
 
+bool AndroidSAFFilesystemNode::remove() {
+	assert(_safTree != nullptr);
+
+	if (!_safNode) {
+		return false;
+	}
+
+	if (!_safParent) {
+		// It's the root of the tree: we can't delete it
+		return false;
+	}
+
+	if (isDirectory()) {
+		// Don't delete folders (yet?)
+		return false;
+	}
+
+	JNIEnv *env = JNI::getEnv();
+
+	bool result = env->CallBooleanMethod(_safTree, _MID_removeNode, _safNode);
+
+	if (env->ExceptionCheck()) {
+		LOGE("SAFFSTree::removeNode failed");
+
+		env->ExceptionDescribe();
+		env->ExceptionClear();
+
+		return false;
+	}
+
+	if (!result) {
+		return false;
+	}
+
+	env->DeleteGlobalRef(_safNode);
+	_safNode = nullptr;
+
+	// Create the parent node to fetch informations needed to make us a non-existent node
```

Comment:
I am unsure I completely understand what we're doing starting from here. The comment helps a bit, but still maybe a bit more clarification needed?